### PR TITLE
Report Storage Sync

### LINKED ISSUE #434: ReportHub
-Objective: Compute ReportVariable and return IDataCube<ReportVariable>
-Required Data Hubs: ReferenceDataHub(only some type (ReportingNode, others?)), ParameterHub(only FxParameter?),IfrsVariableHub)
-Computation: it requires ReportStorage (+called methods) and ReportScopes

Test:
Compute ReportVariable DataCube for a simple reportType (PV). 
Assert that the properties of the report variables are correctly populated (check the join, a prop from IfrsVar and one from GoC)
Compute ReportVariable DataCube for a complex ReportType (PnL)
Assert existence of variables.

### LINKED ISSUE #443: Reconcile Report Storage Data with OpenSmC env

### PR DESCRIPTION
**Pull request by calderinisystemorph**

### CODE REVIEW

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReferenceDataHub.Test/ReferenceDataTest.cs [hunk 1]
```diff
@@ -51,7 +51,7 @@ protected override MessageHubConfiguration ConfigureHost(MessageHubConfiguration
             .ConfigureReferenceDataImportHub();
     }
 
-    [Fact]
+    [Fact (Skip = "Import Hub configuration not working yet.")]
```
Comment (dcolleoni) [inline OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReferenceDataHub.Test/ReferenceDataTest.cs]: undo

**Comment by dcolleoni** [comment]
Rename ClassLibrary directory

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.CalculationScopes/Placeholder/HierarchicalDimensionCacheWithWorkspace.cs
Comment (dcolleoni) [inline OpenSmc.Ifrs17/OpenSmc.Ifrs17.CalculationScopes/Placeholder/HierarchicalDimensionCacheWithWorkspace.cs:1]: Move these files out from CalculationScopes/Placeholder

File: OpenSmc.Ifrs17/ClassLibrary1/OpenSmc.Ifrs17.ImportHubs.csproj
Comment (dcolleoni) [inline OpenSmc.Ifrs17/ClassLibrary1/OpenSmc.Ifrs17.ImportHubs.csproj:1]: Rename ClassLibrary1.
Why do we need all this configuration on IFRSVariable? It should be removed as ReportHub only looks at the "golden" copy of IfrsVariableDataHub

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs [hunk 2]
```diff
@@ -50,17 +51,22 @@ public static MessageHubConfiguration ConfigureReportDataHub(this MessageHubConf
...
                         .WithType<InsurancePortfolio>().WithType<GroupOfInsuranceContract>()
                         .WithType<ReinsurancePortfolio>().WithType<GroupOfReinsuranceContract>())
-                    .FromHub(parameterAddress, ds => ds 
-                        .WithType<ExchangeRate>().WithType<CreditDefaultRate>().WithType<PartnerRating>())
-                    //.FromHub(ifrsVarAddress, ds => ds
-                    //    .WithType<IfrsVariable>())
-                    //.AddCustomInitialization(ReportInit(config)))
-            )
+                    .FromHub(parameterAddress, ds => ds
+                        .WithType<ExchangeRate>().WithType<CreditDefaultRate>().WithType<PartnerRating>()),
```
Comment (dcolleoni) [inline OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs]: only ExchangeRate is needed

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs [hunk 3]
```diff
@@ -73,18 +79,20 @@ public static MessageHubConfiguration ConfigureReportDataHub(this MessageHubConf
...
             using (var universe = scopeFactory.ForSingleton().WithStorage(storage).ToScope<IUniverse>())
             {
                 // TODO: take from the scopes the report variables we need
                 //universe.GetScopes Identities
                 //universe.GetScopes RiskAdjustments
 
-                var reportVariables = new ReportVariable[] { new() { AmountType = "a" }, new() { AmountType = "b" } };
-
-                workspace.Change(new UpdateDataRequest(reportVariables));
+                res = new ReportVariable[] { new() { AmountType = "a" }, new() { AmountType = "b" } };
```
Comment (dcolleoni) [inline OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs]: Call scopes here

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.sln
Comment (dcolleoni) [inline OpenSmc.Ifrs17/OpenSmc.Ifrs17.sln:1]: this file could be safely reset as ImportHubs proj can be removed

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReferenceDataHub.Test/ReferenceDataTest.cs (hunk 1 above)
Comment (amuolo) [reply OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReferenceDataHub.Test/ReferenceDataTest.cs]: undone

File: OpenSmc.Ifrs17/ClassLibrary1/OpenSmc.Ifrs17.ImportHubs.csproj
Comment (amuolo) [reply OpenSmc.Ifrs17/ClassLibrary1/OpenSmc.Ifrs17.ImportHubs.csproj:1]: this is a relic, now I have removed it

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.sln
Comment (amuolo) [reply OpenSmc.Ifrs17/OpenSmc.Ifrs17.sln:1]: done

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs (hunk 2 above)
Comment (amuolo) [reply OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs]: done

**Review by dcolleoni — APPROVED** [review]

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.CalculationScopes/Placeholder/HierarchicalDimensionCacheWithWorkspace.cs
Comment (amuolo) [reply OpenSmc.Ifrs17/OpenSmc.Ifrs17.CalculationScopes/Placeholder/HierarchicalDimensionCacheWithWorkspace.cs:1]: removed

File: OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs (hunk 3 above)
Comment (amuolo) [reply OpenSmc.Ifrs17/OpenSmc.Ifrs17.ReportHub/ReportHubConfiguration.cs]: done

**Pull request merged**